src/connection.js

```diff
@@ -249,6 +249,7 @@ class Connection extends EventEmitter {
     this.socketError = this.socketError.bind(this);
     this.requestTimeout = this.requestTimeout.bind(this);
     this.connectTimeout = this.connectTimeout.bind(this);
+    this.cancelTimeout = this.cancelTimeout.bind(this);
```

Comment:
I'll see what I can do. Might be a bit contrived but at the very least it would be good to test the various types of cancellation